test/tasty/Data/Aeson/CommitTest.hs

```diff
@@ -8,37 +8,67 @@ import           Control.Applicative
 import           Data.Aeson.Commit
 import           Data.Aeson.QQ
 import           Data.Aeson.Types
+import           Data.Foldable       (toList)
 import           Data.Text           (Text)
 import           Test.Tasty.Hspec
 
 tests :: Spec
-tests = testParserWithCases pNested
-  [ ( "fails"
-    , [aesonQQ| {} |]
-    , Left $ unlines
-      [ "Error in $: No match,"
-      , "- key \"value\" not present"
-      , "- key \"nested\" not present"
-      ]
-    )
-  , ( "succeeds unnested"
-    , [aesonQQ| { value: "top" } |]
-    , Right "top"
-    )
-  , ( "succeeds and prefers nested"
-    , [aesonQQ| { value: "top" , nested: { value: "nest" } } |]
-    , Right "nest"
-    )
-  , ( "fails on malformed nested"
-    , [aesonQQ| { value: "top", nested: { foo: 9 } } |]
-    , Left "Error in $.nested: key \"value\" not present"
-    )
-  , ( "fails on nested type mismatch"
-    , [aesonQQ| { value: "top", nested: 9 } |]
-    , Left "Error in $.nested: parsing nestedObj failed, expected Object, but encountered Number"
-    )
-  ]
+tests = do
+  testParserWithCases pNested
+    [ ( "fails"
+      , [aesonQQ| {} |]
+      , Left $ unlines
+        [ "Error in $: No match,"
+        , "- key \"nested\" not present"
+        , "- key \"value\" not present"
+        ]
+      )
+    , ( "succeeds unnested"
+      , [aesonQQ| { value: "top" } |]
+      , Right "top"
+      )
+    , ( "succeeds and prefers nested"
+      , [aesonQQ| { value: "top" , nested: { value: "nest" } } |]
+      , Right "nest"
+      )
+    , ( "fails on malformed nested"
+      , [aesonQQ| { value: "top", nested: { foo: 9 } } |]
+      , Left "Error in $.nested: key \"value\" not present"
+      )
+    , ( "fails on nested type mismatch"
+      , [aesonQQ| { value: "top", nested: 9 } |]
+      , Left "Error in $.nested: parsing nestedObj failed, expected Object, but encountered Number"
+      )
+    ]
+  testParserWithCases parser2
+    [ ("fails with relative path"
+      , [aesonQQ| {"foo": {}} |]
+      , Left $ unlines
+        [ "Error in $: No match,"
+        , "- parsing array failed, expected Array, but encountered Object"
+        , "- .foo: key \"bar\" not present"
+        ]
+      )
+    , ("fails with nested relative path"
+      , [aesonQQ| {"foo": {"bar": ["hello"]}} |]
+      , Left $ unlines
+        [ "Error in $: No match,"
+        , "- parsing array failed, expected Array, but encountered Object"
+        , "- .foo.bar[0]: parsing Int failed, expected Number, but encountered String"
+        ]
+      )
+    , ("fails after commitment"
+      , [aesonQQ| [{}] |]
+      , Left "Error in $: parsing Int failed, expected Number, but encountered Object"
+      )
+    ]
```

Comment:
Yes, I'll add that test as well! Thank you!

Reply:
Thank you for the test. Unfortunately it made me realize that `aeson` is broken in yet another way when dealing with parse errors. The `parserThrowError` takes a relative path and there's no way to pass it an absolute path. Thus extracting the common prefix doesn't work out.

Reply:
```Haskell
-- | Throw a parser error with an additional path.
--
-- @since 1.2.1.0
parserThrowError :: JSONPath -> String -> Parser a
parserThrowError path' msg = Parser $ \path kf _ks ->
    kf (reverse path ++ path') msg
```

Reply:
For the thing to work out nicely I would need somthing like:
```Haskell
parserThrowError' :: JSONPath -> String -> Parser a
parserThrowError' path' msg = Parser $ \_path kf _ks -> kf path' msg
```